lib/pub_sub/breaker.rb

```diff
@@ -0,0 +1,55 @@
+module PubSub
+
+  class Breaker
+
+    THREAD_LOCAL_IDENTIFIER = :pub_sub_region
+
+    class << self
+
+      def run(&block)
+        current_breaker.run(&block)
+      rescue CB2::BreakerOpen
+        Breaker.use_next_breaker
+        sleep 1
+        retry
+      end
+
+      def current_breaker
+        all_breakers[current_breaker_idx]
+      end
+
+      def current_region
+        all_regions[current_breaker_idx]
+      end
+
+      def use_next_breaker
+        Thread.current[THREAD_LOCAL_IDENTIFIER] += 1
+        Thread.current[THREAD_LOCAL_IDENTIFIER] = Thread.current[THREAD_LOCAL_IDENTIFIER] % all_breakers.count
+      end
+
+      private
+
+      def current_breaker_idx
+        Thread.current[THREAD_LOCAL_IDENTIFIER] ||= 0
+      end
+
+      def all_breakers
+        @breakers ||= all_regions.map do |region|
```

Comment:
Just a note: 

I haven't really thought through this extensively and haven't looked into the CB2:Breakers instantiation implementation,  but this memoization may be thread UNSAFE if two threads reach this method at the same exact time. The test below wouldn't quite expose that since `all_breakers` is stubbed.